app/src/main/java/org/android/go/sopt/signup/SignupViewModel.kt

```diff
@@ -0,0 +1,77 @@
+package org.android.go.sopt.signup
+
+import android.content.Context
+import android.text.Editable
+import android.text.TextWatcher
+import android.util.Log
+import android.widget.Toast
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import org.android.go.sopt.ServicePool
+import org.android.go.sopt.data.RequestSignUpDto
+import org.android.go.sopt.data.ResponseSignUpDto
+import retrofit2.Call
+import retrofit2.Response
+
+class SignupViewModel : ViewModel() {
+    private val _signUpResult : MutableLiveData<ResponseSignUpDto> = MutableLiveData()
+    val signUpResult : LiveData<ResponseSignUpDto> = _signUpResult
+
+    private val _errorResult: MutableLiveData<ResponseSignUpDto> = MutableLiveData()
+    val errorResult: LiveData<ResponseSignUpDto> = _errorResult
+
+    private val signUpService = ServicePool.signUpService
+
+    val id = MutableLiveData<String>()
+    val password = MutableLiveData<String>()
+    val name = MutableLiveData<String>()
+    val skill = MutableLiveData<String>()
+
+    val idCheck = MutableLiveData<Boolean>()
+    val pwCheck = MutableLiveData<Boolean>()
+
+    fun checkId(id: String){
+        if(id.isNullOrEmpty() || id!!.matches(Regex("[a-zA-Z0-9]{6,10}"))){
+            idCheck.value = true
+        }else{
+            idCheck.value = false
+        }
+    }
+
+    fun checkPw(password: String){
+        if(password.isNullOrEmpty() || password!!.matches(Regex("^(?=.*[A-Za-z])(?=.*\\d)(?=.*[$@$!%*#?&])[A-Za-z\\d$@$!%*#?&]{6,12}$"))){
+            pwCheck.value = true
+        }else{
+            pwCheck.value = false
+        }
+    }
+
+    fun completeSignUp(id: String, password: String, skill: String, name: String) {
+        signUpService.signup(
+            RequestSignUpDto(
+                id,
+                password,
+                skill,
+                name
```

Comment:
여기는 databinding을 사용하면 completeSignUp에 파라미터들을 받아올 필요가 없을 것 같아용
```suggestion
                name.value.toString()
```

Reply:
아마 아래에 @{data.id} 이런식으로 작성해서 값을 불러오지 못한 것 같습니다!!